# Class 8

## library

In [1]:
import pandas as pd

## load data

In [2]:
data = pd.read_csv("http://raptor.kent.ac.uk/~ds756/data/heart.csv")

In [3]:
# How many rows does it have?
# How many features does it have?
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 918 entries, 0 to 917
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Age             918 non-null    int64  
 1   Sex             918 non-null    object 
 2   ChestPainType   918 non-null    object 
 3   RestingBP       918 non-null    int64  
 4   Cholesterol     918 non-null    int64  
 5   FastingBS       918 non-null    int64  
 6   RestingECG      918 non-null    object 
 7   MaxHR           918 non-null    int64  
 8   ExerciseAngina  918 non-null    object 
 9   Oldpeak         918 non-null    float64
 10  ST_Slope        918 non-null    object 
 11  HeartDisease    918 non-null    int64  
dtypes: float64(1), int64(6), object(5)
memory usage: 86.2+ KB


- How many rows does it have? **918**
- How many features does it have? **12**
- How many features are of type “int64”? **6**

In [4]:
# What is the mean Oldpeak?:
# What is the mean RestingBP?
# What is the mean Age?

data.describe()

Age   RestingBP  Cholesterol   FastingBS       MaxHR  \
count  918.000000  918.000000   918.000000  918.000000  918.000000   
mean    53.510893  132.396514   198.799564    0.233115  136.809368   
std      9.432617   18.514154   109.384145    0.423046   25.460334   
min     28.000000    0.000000     0.000000    0.000000   60.000000   
25%     47.000000  120.000000   173.250000    0.000000  120.000000   
50%     54.000000  130.000000   223.000000    0.000000  138.000000   
75%     60.000000  140.000000   267.000000    0.000000  156.000000   
max     77.000000  200.000000   603.000000    1.000000  202.000000   

          Oldpeak  HeartDisease  
count  918.000000    918.000000  
mean     0.887364      0.553377  
std      1.066570      0.497414  
min     -2.600000      0.000000  
25%      0.000000      0.000000  
50%      0.600000      1.000000  
75%      1.500000      1.000000  
max      6.200000      1.000000

- What is the mean Oldpeak? **0.887364**
- What is the mean RestingBP? **132.396514**
- What is the mean Age? **53.510893**

In [5]:
# What percentage of the cases developed heart disease
heart_cases = data["HeartDisease"].sum()
percentage_heart_cases = (heart_cases / data.shape[0]) * 100
percentage_heart_cases

55.33769063180828

- What percentage of the cases developed heart disease?: **55.33769063180828**

## Index the age feature. 

In [6]:
# index the age feature
data.set_index('Age', inplace=True)

## Create a subset dataframe containing only the data of people in their  40’s.

In [7]:
subset_40s = data[(data.index >= 40) & (data.index < 50)]
subset_40s.describe()

RestingBP  Cholesterol   FastingBS       MaxHR     Oldpeak  \
count  211.000000   211.000000  211.000000  211.000000  211.000000   
mean   127.355450   218.805687    0.123223  145.763033    0.581043   
std     15.749277    86.986344    0.329474   24.012313    0.916474   
min     95.000000     0.000000    0.000000   80.000000   -2.600000   
25%    118.000000   197.000000    0.000000  127.500000    0.000000   
50%    128.000000   233.000000    0.000000  146.000000    0.000000   
75%    139.000000   266.000000    0.000000  167.500000    1.000000   
max    180.000000   491.000000    1.000000  194.000000    3.600000   

       HeartDisease  
count    211.000000  
mean       0.402844  
std        0.491636  
min        0.000000  
25%        0.000000  
50%        0.000000  
75%        1.000000  
max        1.000000

In [8]:
# Compute the required means for the subset
mean_oldpeak = subset_40s['Oldpeak'].mean()
mean_restingbp = subset_40s['RestingBP'].mean()

# Compute the mean of the Age by resetting the index temporarily
mean_age = subset_40s.index.values.mean()

print(f"Mean Oldpeak for people in their 40s: {mean_oldpeak}")
print(f"Mean RestingBP for people in their 40s: {mean_restingbp}")
print(f"Mean Age for people in their 40s: {mean_age}")

Mean Oldpeak for people in their 40s: 0.581042654028436
Mean RestingBP for people in their 40s: 127.35545023696683
Mean Age for people in their 40s: 44.796208530805686


With your 40’s subset compute:
- mean Oldpeak: **0.581043**
- mean RestingBP: **127.355450**
- mean Age: **44.796208530805686**

## Another way to answer

In [9]:
# 1. read data
df = pd.read_csv("http://raptor.kent.ac.uk/~ds756/data/heart.csv")

# 2. Number of rows
num_rows = df.shape[0]
print(f"Number of rows: {num_rows}")

# 3. Number of features
num_features = df.shape[1]
print(f"Number of features: {num_features}")

# 4. Number of features of type 'int64'
num_int64_features = df.select_dtypes(include=['int64']).shape[1]
print(f"Number of features of type 'int64': {num_int64_features}")
print("----------------------------------------------------------------")
# 5. Mean of 'Oldpeak'
mean_oldpeak = df['Oldpeak'].mean()
print(f"Mean Oldpeak: {mean_oldpeak}")

# 6. Mean of 'RestingBP'
mean_resting_bp = df['RestingBP'].mean()
print(f"Mean RestingBP: {mean_resting_bp}")

# 7. Mean of 'Age'
mean_age = df['Age'].mean()
print(f"Mean Age: {mean_age}")
print("----------------------------------------------------------------")
# 8. Percentage of cases that developed heart disease
percentage_heart_disease = (df['HeartDisease'].sum() / num_rows) * 100
print(f"Percentage of cases that developed heart disease: {percentage_heart_disease}%")

print("----------------------------------------------------------------")
# 9. Index the 'Age' feature
df.set_index('Age', inplace=True)
num_features = df.shape[1]
print(df.index)  # Displaying the index after setting it to 'Age'
print("----------------------------------------------------------------")
# 10. Create a subset dataframe containing only the data of people in their 40's
df_40s = df.loc[(df.index >= 40) & (df.index < 50)]

print("Number of people in between 40 and 49 years subset")
# a. Mean Oldpeak for people in their 40's
mean_oldpeak_40s = df_40s['Oldpeak'].mean()
print(f"Mean Oldpeak for people in their 40's: {mean_oldpeak_40s}")

# b. Mean RestingBP for people in their 40's
mean_resting_bp_40s = df_40s['RestingBP'].mean()
print(f"Mean RestingBP for people in their 40's: {mean_resting_bp_40s}")

# c. Mean Age for people in their 40's
mean_age_40s = df_40s.index.values.mean()
print(f"Mean Age for people in their 40's: {mean_age_40s}")

Number of rows: 918
Number of features: 12
Number of features of type 'int64': 6
----------------------------------------------------------------
Mean Oldpeak: 0.8873638344226579
Mean RestingBP: 132.39651416122004
Mean Age: 53.510893246187365
----------------------------------------------------------------
Percentage of cases that developed heart disease: 55.33769063180828%
----------------------------------------------------------------
Index([40, 49, 37, 48, 54, 39, 45, 54, 37, 48,
       ...
       63, 63, 41, 59, 57, 45, 68, 57, 57, 38],
      dtype='int64', name='Age', length=918)
----------------------------------------------------------------
Number of people in between 40 and 49 years subset
Mean Oldpeak for people in their 40's: 0.581042654028436
Mean RestingBP for people in their 40's: 127.35545023696683
Mean Age for people in their 40's: 44.796208530805686
